pennylane_lightning/lightning_gpu/_state_vector.py

```diff
@@ -322,11 +329,19 @@ def _apply_lightning(
             elif method is not None:  # apply specialized gate
                 param = operation.parameters
                 method(wires, invert_param, param)
-            elif isinstance(operation, qml.ops.Controlled) and isinstance(
-                operation.base, qml.GlobalPhase
-            ):  # apply n-controlled gate
-                # LGPU do not support the controlled gates except for GlobalPhase
+            elif (
+                isinstance(operation, qml.ops.Controlled) and not self._mpi_handler.use_mpi
+            ):  # MPI backend does not have native controlled gates support
                 self._apply_lightning_controlled(operation)
+            elif (
+                self._mpi_handler.use_mpi
+                and isinstance(operation, qml.ops.Controlled)
+                and isinstance(operation.base, qml.GlobalPhase)
+            ):
+                # TODO: To move this line to the _apply_lightning_controlled method once the MPI backend supports controlled gates natively
+                raise DeviceError(
+                    "Lightning-GPU-MPI does not support Controlled GlobalPhase gates."
+                )
```

Comment:
Why do we need this condition where we can support `C(GP)` with LGPU-MPI via its matrix?
 ```suggestion
```

Reply:
As we discussed last time, C(GP) is not supported in LGPU-MPI and it's not tested in our pytests. We can't get it support because MPI backend does not suppprt a gate with all wires target.

Reply:
ah, my bad! I saw this custom patch and wanted to get rid of it asap.. :grimacing: